File: GameLibrary/src/cc/lib/checkerboard/Shashki.java
package cc.lib.checkerboard;

/**
 * Also known as Russian Checkers
 */
public class Shashki extends Checkers {
    @Override
    public boolean canJumpSelf() {
        return false;
    }

    @Override
    public boolean canMenJumpBackwards() {
        return true;
    }

    @Override
    public boolean isJumpsMandatory() {
        return true;
    }

    @Override
    public boolean isMaxJumpsMandatory() {
        return true;
    }

    @Override
    public boolean isCaptureAtEndEnabled() {
        return false;
    }

    @Override
    public boolean isFlyingKings() {
        return true;
    }

    @Override
    public boolean isKingPieces() {
        return true;
    }
}

File: Checkerboard/src/cc/lib/checkerboard/Shashki.kt
package cc.lib.checkerboard

/**
 * Also known as Russian Checkers
 */
class Shashki : Checkers() {
	override fun canJumpSelf(): Boolean {
		return false
	}

	override fun canMenJumpBackwards(): Boolean {
		return true
	}

	override val isJumpsMandatory: Boolean
		get() = true
	override val isMaxJumpsMandatory: Boolean
		get() = true
	override val isCaptureAtEndEnabled: Boolean
		get() = false
	override val isFlyingKings: Boolean
		get() = true
	override val isKingPieces: Boolean
		get() = true
}